contrib/drivers/filesharedrivers/nfs/nfs.go

```diff
@@ -163,28 +165,59 @@ func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (*model.FileShareS
 		return nil, err
 	}
 
-	if err := d.cli.CreateVolume(name, vg, opt.GetSize()); err != nil {
-		return nil, err
-	}
-	// remove created volume if got error
-	defer func() {
-		// using return value as the error flag
-		if fshare == nil {
-			if err := d.cli.Delete(name, vg); err != nil {
-				log.Error("failed to remove volume fileshare:", err)
-			}
+	if opt.SnapshotId != "" {
+		// User requested for creating fileshare using existing snapshot
+		//get fileshare name
+		var existingFsName = opt.GetMetadata()[KFileshareName]
+
+		// get existingfileshare snap logical path
+		//get volume group
+		var vg = opt.GetPoolName()
+		// create a fileshare device path
+		var lvPathForSnap = path.Join("/dev", vg, opt.SnapshotName)
+		// create a existing fileshare device path
+		var lvPathExistingPath = path.Join("/dev", vg, existingFsName)
+		// get directory where fileshare mounted
+		var olddirName = path.Join(MountPath, existingFsName)
+		// umount the volume to directory
+		if err := d.cli.UnMount(olddirName); err != nil {
+			log.Error("failed to mount a directory:", err)
```

Comment:
it should be unmount
```suggestion
			log.Error("failed to unmount a directory:", err)
```

Reply:
This is just message to understand the user, actual function is umount only.

Reply:
done